### PR TITLE
Feature/tao 9126/integration tests

### PR DESCRIPTION
**Pull request by oatymart**
I'm adding several integration tests for the `src/git.js` module, using local files and repos.

```
npm run test:integ
npm run test:cov
```

Please note that the target branch should change if we want to merge #35 to develop first.

### CODE REVIEW

File: tests/integration/git/test.js [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+ * Lifecycle functions
+ */
+const setUp = async function setUp() {
+
+    tearDown();
+    try {
+        // copy 3 folders from `fixtures` into work dir
+        fs.copySync(localRepoFixturePath, localRepoPath);
+        fs.copySync(remoteRepoFixturePath, remoteRepoPath); // empty
+        fs.copySync(secondLocalRepoFixturePath, secondLocalRepoPath); // empty
```
Comment (krampstudio) [inline tests/integration/git/test.js]: Is it on purpose than you perform a sync copy ?

```js
await fs.copy(localRepoFixturePath, localRepoPath);
await fs.copyremoteRepoFixturePath, remoteRepoPath);
await fs.copy(secondLocalRepoFixturePath, secondLocalRepoPath);
```

**Comment by oatymart** [comment]
@krampstudio I made a correction. The folders `remoteRepo` and `secondLocalRepo` will now be created in `work` at runtime. Whereas `localRepo` will still be copied across from `fixtures`, to preserve its content.

@ricproenca-tao could you check if this solves your issue, and do you still have an issue with the `signTags` check on Windows?

File: tests/integration/git/test.js (hunk 1 above)
Comment (oatymart) [reply tests/integration/git/test.js]: I know it *should* be equivalent to using `await fs.copy()`, but for some reason it all fails when I replace the synchronous `fs` calls with async ones.

The current way may be a mix of paradigms but it does run.

**Comment by krampstudio** [comment]
I now have another issue : 

```
 hasSignKey
    ✖  no signing key is set up at Test.test (/home/bertrand/dev/projects/tao-extension-release/tests/integration/git/test.js:243:7)
        truefalse
        emptied workDir
Error: spawn git ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:240:19)
    at onErrorNT (internal/child_process.js:415:16)
    at process._tickCallback (internal/process/next_tick.js:63:19)
(node:19137) UnhandledPromiseRejectionWarning: Error: Error: spawn git ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:240:19)
    at onErrorNT (internal/child_process.js:415:16)
    at process._tickCallback (internal/process/next_tick.js:63:19)
    at /home/bertrand/dev/projects/tao-extension-release/node_modules/simple-git/promise.js:62:29
    at Git.<anonymous> (/home/bertrand/dev/projects/tao-extension-release/node_modules/simple-git/src/git.js:632:18)
    at Function.Git.fail (/home/bertrand/dev/projects/tao-extension-release/node_modules/simple-git/src/git.js:1474:18)
    at fail (/home/bertrand/dev/projects/tao-extension-release/node_modules/simple-git/src/git.js:1432:20)
    at /home/bertrand/dev/projects/tao-extension-release/node_modules/simple-git/src/git.js:1441:16
(node:19137) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:19137) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
    ✖  plan != count at process.<anonymous> (/home/bertrand/dev/projects/tao-extension-release/node_modules/tape/index.js:88:19)
        12
        emptied workDir
    ✖  test exited without ending at process.<anonymous> (/home/bertrand/dev/projects/tao-extension-release/node_modules/tape/index.js:88:19)
        undefinedundefined
        emptied workDir
    ✖  test exited without ending at process.<anonymous> (/home/bertrand/dev/projects/tao-extension-release/node_modules/tape/index.js:88:19)
        undefinedundefined
        emptied workDir
    ✖  test exited without ending at process.<anonymous> (/home/bertrand/dev/projects/tao-extension-release/node_modules/tape/index.js:88:19)
        undefinedundefined
        emptied workDir
    ✖  test exited without ending at process.<anonymous> (/home/bertrand/dev/projects/tao-extension-release/node_modules/tape/index.js:88:19)
        undefinedundefined
        emptied workDir
    ✖  test exited without ending at process.<anonymous> (/home/bertrand/dev/projects/tao-extension-release/node_modules/tape/index.js:88:19)
        undefinedundefined
        emptied workDir
    ✖  test exited without ending at process.<anonymous> (/home/bertrand/dev/projects/tao-extension-release/node_modules/tape/index.js:88:19)
        undefinedundefined
        emptied workDir
    ✖  test exited without ending at process.<anonymous> (/home/bertrand/dev/projects/tao-extension-release/node_modules/tape/index.js:88:19)
        undefinedundefined
        emptied workDir
    ✖  test exited without ending at process.<anonymous> (/home/bertrand/dev/projects/tao-extension-release/node_modules/tape/index.js:88:19)
        undefinedundefined
...
```

**Comment by oatymart** [comment]
I switched the order of statements in the failing test. Could you run it again?
Not sure yet if it's platform-dependant. If it still fails after this, I might drop that test from the suite altogether.

**Pull request merged**